src/lib/adapters/CachingAdapter.js

```diff
@@ -0,0 +1,129 @@
+import Tree from '../Tree'
+import Adapter from '../Adapter'
+
+export default class CachingAdapter {
+  constructor(server) {
+    this.highestId = 0
+  }
+
+  async getBookmarksTree() {
+    return this.bookmarksCache
```

Comment:
I do not see how bookmarksCache is getting set.

Reply:
The adapter that inherits from the caching adapter implements onSyncStart and onSyncComplete in order to set the cache. I could also add a method for setting the cache... *shrug*